parsl/db_logger.py

```diff
@@ -0,0 +1,74 @@
+import logging
+
+try:
+    from cmreslogging.handlers import CMRESHandler
+except ImportError:
+    _es_logging_enabled = False
+else:
+    _es_logging_enabled = True
+
+
+class OptionalModuleMissing(Exception):
+    ''' Error raised a required module is missing for a optional/extra provider
+    '''
+
+    def __init__(self, module_names, reason):
+        self.module_names = module_names
+        self.reason = reason
+
+    def __repr__(self):
+        return "Unable to Initialize logger.Missing:{0},  Reason:{1}".format(
+            self.module_names, self.reason
+        )
+
+
+class NullHandler(logging.Handler):
+    """Setup default logging to /dev/null since this is library."""
+
+    def emit(self, record):
+        pass
+
+
+def get_db_logger(host='search-parsl-logging-test-2yjkk2wuoxukk2wdpiicl7mcrm.us-east-1.es.amazonaws.com',
+                  port=443,
+                  level=logging.CRITICAL,
+                  index_name="parsl.campaign"):
+    """
+    Parameters
+    ----------
+    host : str, optional
+        URL to the elasticsearch cluster. Skip the http(s)://
+    port : int, optional
+        Port to use to access the elasticsearch cluster
+    level : logging.LEVEL, optional
+        This module will log *only* if logging.INFO or lower is provided.
+    index_name : str, optional
+        Index name to use for elasticsearch
+
+    Returns
+    -------
+    logging.logger object
+
+    Raises
+    ------
+
+    """
+    logger = logging.getLogger("ParslElasticsearch")
```

Comment:
I favor logger = logging.getLogger(__file__), to be consistent with other modules

Reply:
We should be comparing this to the get_*_logger methods in parsl/__init__.py where we name the logger "parsl". This doesn't really add much anyway.